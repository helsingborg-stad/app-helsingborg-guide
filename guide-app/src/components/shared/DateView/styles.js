// @flow

import { StyleSheet } from "react-native";
import { Colors, TextStyles } from "@assets/styles";
import { StyleSheetUtils } from "@utils";

export default StyleSheet.create({
  text: StyleSheetUtils.flatten([
    TextStyles.description,
    {
      color: Colors.gray3
    }
  ])
});
