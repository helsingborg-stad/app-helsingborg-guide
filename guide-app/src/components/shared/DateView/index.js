// @flow

import React from "react";
import { Text } from "react-native";

import styles from "./styles";

type Props = {
  startDate?: ?string,
  endDate?: ?string
};

const DateView = (props: Props) => {
  const { startDate, endDate } = props;
  if (!startDate || !endDate) {
    return null;
  }

  const start = new Date(startDate);
  const end = new Date(endDate);

  try {
    return (
      <Text style={styles.text} numberOfLines={1}>
        {`${start
          .toISOString()
          .substring(0, 10)} - ${end.toISOString().substring(0, 10)}`}
      </Text>
    );
  } catch (error) {
    // failed to convert date to ISO string
  }
  return null;
};

DateView.defaultProps = {
  startDate: null,
  endDate: null
};

export default DateView;
