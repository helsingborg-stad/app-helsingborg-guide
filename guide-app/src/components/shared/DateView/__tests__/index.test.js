// @flow

/* eslint-disable import/first */

import "react-native";
import React from "react";
import DateView from "../index";

import renderer from "react-test-renderer";

const startDate: string = "2018-05-14";
const endDate: string = "2018-08-31";

test("No props", () => {
  const tree = renderer.create(<DateView />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("Only start date", () => {
  const tree = renderer.create(<DateView startDate={startDate} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("Only end date", () => {
  const tree = renderer.create(<DateView endDate={endDate} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("Correct input", () => {
  const tree = renderer
    .create(<DateView startDate={startDate} endDate={endDate} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("Invalid input", () => {
  const tree = renderer
    .create(
      <DateView startDate={startDate} endDate="this is not a date string" />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
