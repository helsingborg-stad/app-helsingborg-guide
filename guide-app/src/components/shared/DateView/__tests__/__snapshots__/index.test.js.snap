// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Correct input 1`] = `
<Text
  numberOfLines={1}
  style={
    Object {
      "color": "#7f7f7f",
      "fontFamily": "Roboto",
      "fontSize": 16,
      "lineHeight": 22,
    }
  }
>
  2018-05-14 - 2018-08-31
</Text>
`;

exports[`Invalid input 1`] = `null`;

exports[`No props 1`] = `null`;

exports[`Only end date 1`] = `null`;

exports[`Only start date 1`] = `null`;
